01_fib/n0npax/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"os"
+)
+
+// Allow to overwtite stream by tests
+var out io.Writer = os.Stdout
+
+// This function is useless and it's provided just due to requirements
+func main() {
+	fib(7)
+}
+
+// Wrapper function to print computed Fibonacci numbers
+func fib(num int) {
+	for _, v := range dispatchFib(num) {
+		fmt.Fprintln(out, v)
+	}
+}
+
+// Dispatch and return proper numbers
+func dispatchFib(num int) []int {
+	switch {
+	case num == 0:
+		return []int{0}
+	case num < 0:
+		return computeNegafibonacci(num)
+	default:
+		return computeFibonacci(num)
+
+	}
+}
+
+// Compute Fibonacci numbers
+func computeFibonacci(num int) []int {
+	fibNumbers := make([]int, num)
+	a, b := 1, 1
+	for i := range fibNumbers {
+		fibNumbers[i] = a
+		a, b = b, a+b
+	}
+	return fibNumbers
+}
+
+// Compute negafibonacci numbers
+func computeNegafibonacci(num int) []int {
+	fibNumbers := computeFibonacci(num * -1)
+	for i, v := range fibNumbers {
+		fibNumbers[i] = v * int(math.Pow(-1, float64(i%2)))
```

Comment:
Since you are already using slices in this code, you could optimise this expression to remove `math.Pow` and the float/int conversions:
```
    fibNumbers[i] = v * [2]int{1, -1}[i%2]
```

i.e. index a static array instead of computing -1^0 and -1^1 each time.

Just an idea to demonstrate a different way. No need to make this change.

Reply:
Good hint. Thanks for this!